### ISSUE TITLE
Upgrade the logic of towers and missiles

### ISSUE THREAD

**Issue by m0rp30** [crystal-bit/defending-todot#8]
Maybe the logic of the towers is okay but I would add more reactivity to the missiles but I would also give it a random error for the contact point of the enemy in order to make the game more dynamic

**Comment by davcri** [comment]
I agree! I'm going to create a PR to have a more responsive turret fire mechanic.  
For the second idea: I'm not quite sure about it. I prefer the gameplay to be predictable and "readable". I think that introducing a random error for the projectile collision can confuse the player.
